common/packet/src/packet.cpp

```diff
@@ -0,0 +1,239 @@
+#include "packet.hpp"
+
+/**
+ * Packet class
+ * Write and Read - byte, int, float, double, glm::vec3
+ * Read returns -1 on error, size of data read otherwise
+ * 
+ * Note: currently no support for checking if read in the wrong order
+ *      or if read type mismatches from original written value
+*/
+
+union Union32 {
+    float f;
+    int i;
+    u_int32_t l;
+} union32;
+
+union Union64 {
+    double f;
+    u_int64_t l;
+} union64;
+
+Packet::Packet() {
+    this->buffer = std::vector<u_int8_t>();
+}
+
+Packet::~Packet() {
+    this->buffer = std::vector<u_int8_t>();
+}
```

Comment:
```suggestion
Packet::Packet() {
    this->buffer = std::vector<u_int8_t>();
}

Packet::~Packet() {
    this->buffer = std::vector<u_int8_t>();
}
```
Unneeded, the default constructor/destructor already handles this. If you're gonna add a custom destructor, see [rule of three](https://en.wikipedia.org/wiki/Rule_of_three_(C%2B%2B_programming)#:~:text=The%20rule%20of%20three%20(also,copy%20assignment%20operator)

Reply:
Good point, thanks